src/Integration/WPCOMProxy.php

```diff
@@ -0,0 +1,209 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Integration;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Registerable;
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Service;
+use WP_REST_Response;
+use WP_REST_Request;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class WPCOMProxy
+ *
+ * Initializes the hooks to filter the data sent to the WPCOM proxy depending on the query parameter gla_syncable.
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Integration
+ */
+class WPCOMProxy implements Service, Registerable {
+
+	/**
+	 * The Post types to be filtered.
+	 *
+	 * @var array
+	 */
+	public static $post_types_to_be_filter = [
+		'product'           => [
+			'meta_query' => [
+				[
+					'key'     => '_wc_gla_visibility',
+					'value'   => 'sync-and-show',
+					'compare' => '=',
+				],
+			],
+		],
+		'shop_coupon'       => [
+			'meta_query' => [
+				[
+					'key'     => '_wc_gla_visibility',
+					'value'   => 'sync-and-show',
+					'compare' => '=',
+				],
+				[
+					'key'     => 'customer_email',
+					'compare' => 'NOT EXISTS',
+				],
+			],
+		],
+		'product_variation' => [
+			'meta_query' => null,
+		],
+	];
+
+	/**
+	 * Register all filters.
+	 */
+	public function register(): void {
+		// Allow to filter by gla_syncable.
+		add_filter(
+			'woocommerce_rest_query_vars',
+			function ( $valid_vars ) {
+				$valid_vars[] = 'gla_syncable';
+				return $valid_vars;
+			}
+		);
+
+		foreach ( array_keys( self::$post_types_to_be_filter ) as $object_type ) {
+			$this->register_object_types_filter( $object_type );
+		}
+	}
+
+	/**
+	 * Register the filters for a specific object type.
+	 *
+	 * @param string $object_type The object type.
+	 */
+	protected function register_object_types_filter( string $object_type ): void {
+		add_filter( 'woocommerce_rest_prepare_' . $object_type . '_object', [ $this, 'check_item_is_syncable' ], 9, 3 );
```

Comment:
For consistency it would be nice if this was written over multiple lines as the two filters added below are.